### PR TITLE
Issue MUI table Recurring api

### PR DESCRIPTION
**Pull request by theobonzi**
This PR enhances the readability of the Recurring API page by integrating an MUI datatable and expanding its width.
Solving issue #4828

### CODE REVIEW

**Comment by theobonzi** [comment]
@Theodlz

**Comment by Theodlz** [comment]
@theobonzi minor comment but when you mention the issue on the PR's description, say something like: Solving issue <issue_number>. Simply because issues and PRs are in the same number sequence, so until I click on it I can't tell if you are solving an issue, or re-opening a PR based on an existing one.

**Comment by Theodlz** [comment]
Unrelated to the review, but when you are ready to have your code reviewed please click on "ready for review" so that it's not a draft anymore, and remove the WIP unless you have code to add

File: static/js/components/RecurringAPIPage.jsx [hunk 1]
```diff
@@ -156,47 +163,191 @@ const RecurringAPIPage = () => {
...
+    </div>
   );
 };
 
-RecurringAPIList.propTypes = {
-  // eslint-disable-next-line react/forbid-prop-types
-  recurringAPIs: PropTypes.arrayOf(PropTypes.any).isRequired,
-  deletePermission: PropTypes.bool.isRequired,
-};
+//   return (
```
Comment (Theodlz) [inline static/js/components/RecurringAPIPage.jsx:318]: remove commented code unless it's a mistake

**Review by Theodlz — CHANGES_REQUESTED** [review]
Very confused here, most of it is just ... commented code??? Make sure to not leave all of that commented stuff when pushing to a PR, unless you need it for later. 

I'm also not sure that all of it should be commented, looks like some of what you removed is actually important.

**Comment by Theodlz** [comment]
@theobonzi please rename your PR without the WIP and make it a normal PR (not a draft) when this is ready for review. Otherwise no way for the reviewer to know!

**Review by Theodlz — APPROVED** [review]
LGTM 🚀

**Comment by Theodlz** [comment]
@just merged your PR @theobonzi! Just one comment for next time, you added "issue" in your PR, which doesn't make much sense. You mention in the description what issue you are solving, but a PR's name is something reflecting the feature you are developing. It's not an issue but a solution!

**Pull request merged**